Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -328,6 +328,38 @@ extension Compiler.ByteCodeGen {
     builder.label(success)
   }
 
+  mutating func emitAtomicNoncapturingGroup(
+    _ child: DSLTree.Node
+  ) throws {
+    /*
+      save(continuingAt: success)
+      save(restoringAt: intercept)
+      <sub-pattern>    // failure restores at intercept
+      clearThrough(intercept) // remove intercept and any leftovers from <sub-pattern>
+      fail             // ->success
+    intercept:
+      clearSavePoint   // remove success
+      fail             // propagate failure
+    success:
+      ...
+    */
```

Comment:
Is the only difference between lookahead and atomic the `continuingAt` vs `restoringAt`?

Reply:
Right, _positive_ lookaheads and atomic groups have the same behavior, except that lookaheads reset the matching position on exit.

Reply:
Should we parameterize and share code in that case?

Reply:
Feel like we could come back and do that, but we'd probably want to extricate the negative lookahead code in that case.